main.go

```diff
@@ -349,9 +353,16 @@ func runWithOptions(opts Options, out output, logger *zap.Logger) {
 		out.Fatalf("Failed while preparing the request: %v\n", err)
 	}
 
```

Comment:
I can not comment on the line I would like, but line:342.

Why are we calling `serializer.Request()` for a streaming call? Should we call `serializer.StreamRequest()`?
This will help to not have a mixt usage of stream and unary request with the method `Request`

Reply:
`serializer.Request()` is needed for both unary and stream requests. It holds the metadata like procedure-name, shard-key, headers, timeout, etc. (but not the request-body) which is needed to establish the stream

While `serializer.StreamRequest()` fetches request to be dispatched once the stream has been established